src/libraries/System.Text.Json/gen/src/IsExternalInit.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Dummy class so C# init-only properties can compile on NetStandard.
+    /// </summary>
+    public sealed class IsExternalInit { }
```

Comment:
Nit: file seems like a good candidate for `Common`?

Reply:
Does the type need to be public for that to work or could we make it internal?

Reply:
Just tried it - it can be internal. We have 4 instances of this file in the repo. Would you recommend re-using [corelib's implementation](https://github.com/dotnet/runtime/blob/ccc25a90082fa6592eeca8e146ff37e1b7d6b79d/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/IsExternalInit.cs) and #ifdef-ing the visibility based on the `SYSTEM_PRIVATE_CORELIB` constant? I'll tackle as part of https://github.com/dotnet/runtime/issues/45448#issuecomment-822144624.